### ISSUE TITLE
Feature request: including classes that implement an interface

### ISSUE THREAD

**Issue by clamothe** [vojtechhabarta/typescript-generator#285]
Allow a configuration in which all classes that implement a particular interface are included.

`          ScanResult.getClassesImplementing("com.example.ApiObject")
`

This is handy to use in-hand with polymorphic Jackson:

`
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY)
public interface ApiObject {}
`

I'll create a PR for this in time if no one beats me to it.

**Comment by vojtechhabarta** [comment]
Ok, so do you plan to add `classesImplementingInterfaces` next to `classesWithAnnotations` parameter?

**Comment by clamothe** [comment]
Something like that. The implementation should probably support specifying
either a class or interface, of which all descendents will be included.

**Comment by clamothe** [comment]
@antch That's a great suggestion. 

Could you elaborate on the `getSuperClass()` option?

**Comment by clamothe** [comment]
@antch 

I wasn't able to get `classWithAnnotations` to work as described with an `@Inherited` annotation. I annotated an interface with it, but the implementations of the interface are not included. Perhaps this only works with a class?

**Comment by vojtechhabarta** [comment]
Yes, I tried foillowing code and annotation inheritance only worked when `ApiObject` was abstract class (not interface):
``` java
System.out.println(ApiObject.class.getAnnotation(GenerateTypescript.class));
System.out.println(SomeApiObject.class.getAnnotation(GenerateTypescript.class));
```

I also found this in javadoc of `@Inherited` annotation:
> ... Note also that this meta-annotation only causes annotations to be inherited from superclasses; annotations on implemented interfaces have no effect.

**Comment by clamothe** [comment]
Ah, interesting. That explains that.

**Comment by vojtechhabarta** [comment]
@clamothe anything new? I am cleaning issues after release, can I close this issue or are you still interested in creating PR?

**Comment by clamothe** [comment]
Thanks for the reminder. PR created

**Comment by clamothe** [comment]
Thanks for merging. Would it be possible to have a release that includes this in the next week or two?

**Comment by vojtechhabarta** [comment]
@clamothe I will release it till the end of this week.

**Comment by clamothe** [comment]
Thank you!

**Comment by vojtechhabarta** [comment]
Released in version [2.12.476](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.12.476).
